src/app/Models/Location.php

```diff
@@ -92,6 +92,17 @@ public static function geocodeMissing() {
         return $count;
     }
 
+    /**
+     * Get all the locations with a source $source.
+     *
+     * @return An array of all the locations with a source that matches $source.
+     */
+    public static function locationsWithSource($source) {
```

Comment:
I'd suggest changing this to:

public function scopeBySource($query, $source) {
        return $query->whereHas('locationSource', function ($s) use($source) {
            $s->where('name', '=', $source);
        });
}